lib/redis_client/cluster/router.rb

```diff
@@ -249,10 +249,22 @@ def send_cluster_command(method, command, args, &block)
         end
       end
 
-      def send_script_command(method, command, args, &block)
+      def send_script_command(method, command, args, &block) # rubocop:disable Metrics/AbcSize
         case ::RedisClient::Cluster::NormalizedCmdName.instance.get_by_subcommand(command)
         when 'debug', 'kill'
           @node.call_all(method, command, args, &block).first
+        when 'exists'
+          # check for existence on all nodes
+          responses = @node.call_all(method, command, args, &block)
+          final_response = []
+          i = -1
+          loop do
+            i += 1
+            break if responses[0][i].nil?
+
+            final_response << (responses.all? { |r| r[i].eql?(1) } ? 1 : 0)
```

Comment:
It looks like we can write the logic as a one liner like this:

```
irb(main):013:0> [[1,0],[1,0]].transpose.map { |arr| arr.any?(&:zero?) ? 0 : 1 }
=> [1, 0]

irb(main):014:0> [[1,0],[0,0]].transpose.map { |arr| arr.any?(&:zero?) ? 0 : 1 }
=> [0, 0]

irb(main):015:0> [[1,0],[0,1]].transpose.map { |arr| arr.any?(&:zero?) ? 0 : 1 }
=> [0, 0]

irb(main):016:0> [[0,0],[0,0]].transpose.map { |arr| arr.any?(&:zero?) ? 0 : 1 }
=> [0, 0]
```

Reply:
ACK. beautiful.